Fixes for codebook deployment

CBroz1 opened the pull request:
- remove assertion of directory name from notebooks as this won't work on codebook
- add mention of data directory for codebook
- move all export items from `pipeline` to `export`, include warning for users not using `no-curation` `ephys_mode`

kabilar commented on workflow_array_ephys/export.py:
```diff
@@ -13,3 +11,21 @@
+if ephys_mode == "no-curation":
+    from element_array_ephys.export.nwb import ecephys_session_to_nwb, write_nwb
+else:
+    missing_ephys_warning = (
+        f"Warning: ephys export requires the no-curation ephys_mode. To use it,\n\t"
+        + "try setting datajoint.config['custom']['ephys_mode'] to 'no-curation'\n\t"
+        + "and restarting your kernel."
+    )
+
+    def ecephys_session_to_nwb(*args):
+        return missing_ephys_warning
+
+    def write_nwb(*args):
+        return missing_ephys_warning
+
+    print(missing_ephys_warning)
```
```suggestion
    raise NotImplementedError(
        "Ephys export requires the no-curation ephys_mode. To use it,\n\t"
        + "try setting datajoint.config['custom']['ephys_mode'] to 'no-curation'\n\t"
        + "and restarting your kernel."
    )
```

CBroz1 replied:
I didn't want to raise an error here because that would halt the import process, and would prevent a user from trying out the upstream import functions.

My motivation for adding these dummy functions was to appease the linter. 
1. If a function isn't used, as is the case with this didactic script, the linter flags an issue. So, we add it to `__all__`
2. If a function isn't defined, but it is in `__all__`, as would be the case for an `acute` user, the import process throws an error. 

We can move forward with just accepting linter warnings for didactic scripts

kabilar replied:
Thanks for clarifying.  You can ignore my suggestion.

kabilar commented on notebooks/py_scripts/02-workflow-structure-optional.py:
```diff
@@ -7,130 +7,103 @@
+#
+# - `<table>()` show table contents
+# - `heading` shows attribute definitions
+# - `describe()` show table defintiion with foreign key references
```
```suggestion
# - `describe()` show table definition with foreign key references
```

kabilar reviewed (approved):
Thanks @CBroz1.  Two minor comments above.

kabilar commented:
Please update changelog.

CBroz1 commented:
Thanks for your review @kabilar. I've updated the changelog and fixed the parallel typo in the `ipynb`. See comment above regarding changes to `export.py`

kabilar reviewed (approved):
Thanks @CBroz1!
